flyteorg/flytekit
# Remove storage as a task resource option

## Tracking issue
Fixes https://github.com/flyteorg/flyte/issues/3955

## Why are the changes needed?
Make the storage unused in the task resource option, so that users will not be confused about why they can define the storage but it didn't work.

## What changes were proposed in this pull request?
Remove all variables with task resource `storage`.
Search variable like this.
![image](https://github.com/flyteorg/flytekit/assets/76461262/53693760-2d59-43d1-8aa2-f03027794a2e)

## How was this patch tested?
1. run flytekit task with specified resource
2. use `kubectl describe nodeId` to check if the `Resources` as expected
### Success Scenario
```python
from flytekit import task, Resources

@task(
        requests=Resources(cpu="2", ephemeral_storage="200Mi", mem="200Mi"),
)
def test_storage() -> None:
    import time
    print("@@@ I am sleeping for 10 seconds")
    time.sleep(10)
    return
```

```bash
pyflyte run --remote remove_resource_storage.py test_storage
```

```
kubectl describe pod ff436fac5f4d74583b75-f4wb3a3y-0 -n flytesnacks-development
```

The output is below.
```yaml
    Limits:
      cpu:                2
      ephemeral-storage:  200Mi
      memory:             200Mi
    Requests:
      cpu:                2
      ephemeral-storage:  200Mi
      memory:             200Mi
```
![image](https://github.com/flyteorg/flytekit/assets/76461262/706d1677-e883-447e-ab98-6650aa018858)
### Failure Scenario
```python
from flytekit import task, Resources

@task(
        requests=Resources(cpu="2", storage="200Mi"),
)
def test_storage() -> None:
    import time
    print("@@@ I am sleeping for 10 seconds")
    time.sleep(10)
    return
```

```bash
pyflyte run --remote remove_resource_storage.py test_storage
```
![image](https://github.com/flyteorg/flytekit/assets/76461262/c155eef9-aee3-49da-8b85-ee463c18da04)

## Check all the applicable boxes 


## Related PRs
https://github.com/flyteorg/flyte/pull/4658
